test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -0,0 +1,258 @@
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+
+    assert doc_chunks[0].content == "A simple sentence1."
+    assert doc_chunks[0].meta["split_id"] == 0
+    assert doc_chunks[0].meta["split_idx_start"] == 0
+    assert doc_chunks[0].meta["_split_overlap"] == [{"doc_id": doc_chunks[1].id, "range": (0, 11)}]
+
+    assert doc_chunks[1].content == " sentence1. A bright sentence2."
+    assert doc_chunks[1].meta["split_id"] == 1
+    assert doc_chunks[1].meta["split_idx_start"] == 8
+    assert doc_chunks[1].meta["_split_overlap"] == [
+        {"doc_id": doc_chunks[0].id, "range": (8, 19)},
+        {"doc_id": doc_chunks[2].id, "range": (0, 11)},
+    ]
+
+    assert doc_chunks[2].content == " sentence2. A clever sentence3."
+    assert doc_chunks[2].meta["split_id"] == 2
+    assert doc_chunks[2].meta["split_idx_start"] == 28
+    assert doc_chunks[2].meta["_split_overlap"] == [
+        {"doc_id": doc_chunks[1].id, "range": (20, 31)},
+        {"doc_id": doc_chunks[3].id, "range": (0, 11)},
+    ]
+
+    assert doc_chunks[3].content == " sentence3. A joyful sentence4"
+    assert doc_chunks[3].meta["split_id"] == 3
+    assert doc_chunks[3].meta["split_idx_start"] == 48
+    assert doc_chunks[3].meta["_split_overlap"] == [{"doc_id": doc_chunks[2].id, "range": (20, 31)}]
+
+
+def test_recursive_splitter_generate_pages():
+    splitter = RecursiveDocumentSplitter(split_length=18, split_overlap=0, separators=[" "], keep_separator=True)
+    doc = Document(content="This is some text. \f This text is on another page. \f This is the last page.")
+    doc_chunks = splitter.run([doc])
+    doc_chunks = doc_chunks["documents"]
+    assert len(doc_chunks) == 7
+    for doc in doc_chunks:
+        if doc.meta["split_id"] in [0, 1, 2]:
+            assert doc.meta["page_number"] == 1
+        if doc.meta["split_id"] in [3, 4]:
+            assert doc.meta["page_number"] == 2
+        if doc.meta["split_id"] in [5, 6]:
+            assert doc.meta["page_number"] == 3
+
+
+def test_recursive_splitter_separator_exists_but_split_length_too_small_fall_back_to_character_chunking():
+    splitter = RecursiveDocumentSplitter(separators=[" "], split_length=2)
+    doc = Document(content="This is some text. This is some more text.")
+    result = splitter.run(documents=[doc])
+    assert len(result["documents"]) == 21
+    for doc in result["documents"]:
+        assert len(doc.content) == 2
+
+
+def test_recursive_splitter_generate_empty_chunks():
+    splitter = RecursiveDocumentSplitter(split_length=15, separators=["\n\n", "\n"], keep_separator=False)
+    text = "This is a test.\n\n\nAnother test.\n\n\n\nFinal test."
+    doc = Document(content=text)
+    chunks = splitter.run([doc])["documents"]
+
+    assert chunks[0].content == "This is a test."
+    assert chunks[1].content == "\nAnother test."
+    assert chunks[2].content == "Final test."
+
+
+# def test_recursive_splitter_fallback_to_character_chunking():
```

Comment:
Should this test be reactivated?

Reply:
what I wanted to test here was already being triggered by another test, but I can explicitly add this one